lib/core/presentation/pages/home/home_page.dart

```diff
@@ -1,11 +1,35 @@
+import 'package:app/core/service/firebase/firebase_service.dart';
+import 'package:app/core/utils/navigation_utils.dart';
 import 'package:app/theme/color.dart';
 import 'package:auto_route/auto_route.dart';
+import 'package:firebase_messaging/firebase_messaging.dart';
 import 'package:flutter/material.dart';
 
 @RoutePage()
-class HomePage extends StatelessWidget {
+class HomePage extends StatefulWidget {
   const HomePage({super.key});
 
+  @override
+  State<StatefulWidget> createState() => _HomePageViewState();
+}
+
+class _HomePageViewState extends State<HomePage> {
+  
+  Future<void> setupInteractedMessage() async {
```

Comment:
Please move these setup to _App widget in app.dart file,  these are global service setup, should not be specific in any business screen

Reply:
Got it, thanks 👍